### PR TITLE
Create UserProfile Component (#62)

### PR DESCRIPTION
**Pull request by damdiana**

### CODE REVIEW

File: src/app/profile/layout.tsx [hunk 1]
```diff
@@ -0,0 +1,85 @@
...
+      />
+      <main className="h-screen">
+        <div className="flex mt-4 w-full h-full">
+          <div className="flex flex-col m-2 p-1 border gap-2 w-4/12 h-2/6">
+            <h1 className="text-lg font-bold">My account</h1>
+            <Link
+              href="/profile/settings"
+              className={
+                currentRoute === "/profile/settings"
+                  ? "p-1 bg-black text-white"
```
Comment (iampava) [inline src/app/profile/layout.tsx:52]: We should extract those strings into variables, and re-use them.

File: src/components/UserProfile/UserFavorites.tsx [hunk 2]
```diff
@@ -0,0 +1,16 @@
+import { faHeart } from "@fortawesome/free-solid-svg-icons";
+import { FontAwesomeIcon } from "@fortawesome/react-fontawesome";
+
+function UserFavorites() {
+  return (
```
Comment (iampava) [inline src/components/UserProfile/UserFavorites.tsx:5]: We can put this inside `favorites/page.tsx`. No need for a new component now, that is read only.

File: src/components/EditableField/EditableField.tsx [hunk 3]
```diff
@@ -0,0 +1,99 @@
...
+  const [isEditable, setIsEditable] = useState(false);
+
+  if (isEditable === true) {
+    return (
+      <div>
+        <h2 className="text-lg m-2">
+          Edit your <span className="lowercase"> {label} </span>
+        </h2>
+        <div className="p-1 ">
+          <form className="flex justify-betwee flex-col">
```
Comment (iampava) [inline src/components/EditableField/EditableField.tsx:25]: Are you sure this works when clicking `Enter`?

**Review by iampava — CHANGES_REQUESTED** [review]

**Review by iampava — CHANGES_REQUESTED** [review]

File: src/components/EditableField/EditableField.tsx [hunk 4]
```diff
@@ -0,0 +1,100 @@
...
+import { HTMLInputTypeAttribute, useState } from "react";
+
+type Props = {
+  label: string;
+  initialValue: string;
+  type: HTMLInputTypeAttribute;
+  onSave: (value: string) => void;
+};
+
+const EditableField = ({ label, initialValue, type, onSave }: Props) => {
```
Comment (iampava) [inline src/components/EditableField/EditableField.tsx:13]: Let's add a storybook for this one

File: src/components/UserProfile/UserSettings.tsx [hunk 5]
```diff
@@ -0,0 +1,34 @@
+"use client";
+import { useState } from "react";
+import EditableField from "../EditableField/EditableField";
+import { text } from "stream/consumers";
+
+function UserSettings() {
```
Comment (iampava) [inline src/components/UserProfile/UserSettings.tsx:6]: Remove from `UserProfile` folder. Put it directly in `components.

**Review by iampava — CHANGES_REQUESTED** [review]

File: src/components/EditableField/EditableField.tsx (hunk 4 above)
Comment (damdiana) [reply src/components/EditableField/EditableField.tsx:13]: Storybook currently not working

File: src/components/EditableField.tsx [hunk 6]
```diff
@@ -0,0 +1,100 @@
...
+  if (isEditable === true) {
+    return (
+      <div>
+        <h2 className="text-lg m-2">
+          Edit your <span className="lowercase"> {label} </span>
+        </h2>
+        <div className="p-1 ">
+          <form className="flex justify-betwee flex-col" onSubmit={onSubmit}>
+            <div className="flex border border-black border-solid mb-2 ">
+              <input
```
Comment (iampava) [inline src/components/EditableField.tsx:33]: Shouldn't this be required? Did you check if I can save it while being empty?

File: src/components/UserSettings.tsx [hunk 7]
```diff
@@ -0,0 +1,34 @@
...
+function UserSettings() {
+  const [fullName, setFullName] = useState("Diana Damian");
+  const [email, setEmail] = useState("diana.damian@gmail.com");
+  return (
+    <div className="w-full h-full flex">
+      <div className="m-2 p-1 border w-full h-3/5">
+        <h2 className="text-lg font-thin m-2"> ACCOUNT DETAILS </h2>
+        <div className="p-1">
+          <EditableField
+            label={"Full Name"}
```
Comment (iampava) [inline src/components/UserSettings.tsx:15]: No need to put the `{` if it's just a string. It can be simply `label="Full Name"`.

Do the same for all

**Review by iampava — APPROVED** [review]
2 small changes needed + rename the issue. Afterwards you can merge

**Pull request merged**